lib/chatwithtools.py

```diff
@@ -50,35 +80,65 @@ def __init__(self, runner: ChatLlamaCpp):
         self.model = runner
 
     def _process_single_input(self, input_data: dict[str, Any]) -> dict[str, Any]:
-        model_with_tools = self.model.bind_tools(json.loads(input_data['tools']))
+        system_prompt = """
+{downstream_system_prompt}
```

Comment:
if this is done like this, tool call parsing would need to be done for every new kind of template.
how about the manual way is done for only the models that do not support tool calling.
it can be detected from the model kwargs in config.json. The native supporters would have `tool_choice` set to True.

Reply:
I don't understand the concern. Can you rephrase?

Reply:
ah nevermind this one. I previously thought the prompt for the model was hand-crafted. It turns out it's only the glue code between the task type and langchain wrapper.
All is good.